tests/sdk/v2/test_workflow.py

```diff
@@ -59,8 +63,11 @@ def faked_task_wf():
 
 
 def test_workflow_undecorated_task():
-    with pytest.warns(_workflow.NotATaskWarning):
+    with pytest.warns(_workflow.NotATaskWarning) as warnings:
         _ = undecorated_task_wf()
+    warning = str(warnings[0])
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Can you add an explicit check for a single warning?